app/src/main/java/com/aconno/sensorics/BluetoothScanningService.kt

```diff
@@ -150,11 +150,15 @@ class BluetoothScanningService : DaggerService() {
     private fun startTimer(deviceList: List<Device>? = null) {
         //Launches non-blocking coroutine
         scanTimerDisposable = GlobalScope.launch(context = Dispatchers.Main) {
-            delay(ANDROID_N_MAX_SCAN_DURATION - 60 * 1000)
-            Timber.tag("Sensorics - BLE").d("Stopping")
-            stopScanning()
-            Timber.tag("Sensorics - BLE").d("Restarting")
-            startScan(deviceList)
+            while (true) {
+                delay(ANDROID_N_MAX_SCAN_DURATION - 60 * 1000)
+
+                Timber.tag("Sensorics - BLE").d("Restarting")
+                bluetooth.apply {
```

Comment:
Why `bluetooth.apply`?

Reply:
To make the code more clean. Without using `bluetooth.apply`, this code snippet would look like this:
```
bluetooth.stopScanning()
if(deviceList == null) bluetooth.startScanning() else bluetooth.startScanning(deviceList)

```
or like this:

```
bluetooth.stopScanning()
if(deviceList == null) {
    bluetooth.startScanning()
} else {
    bluetooth.startScanning(deviceList)
}

```

Reply:
Yeah, but how was that working before?

Reply:
There are `startScanning` and `stopScanning` methods in this class (BluetoothScanningService) that have been called in the old version of code. So, now, instead of calling this methods, only bluetooth.stopScanning and bluetooth.startScanning are called. The problem was that BluetoothScanningService.stopScanning clears all resources required to start scanning again and the BluetoothScanningService.startScanning method does not reinitialize all this resources.  So that's why that code had to be changed with this one that does only what has to be need in order to make scanning not stop after 30 minutes - to restart bluetooth scanning, and not to clear the resources that are still needed.